src/components/chip/chip.ts

```diff
@@ -78,6 +78,12 @@ export default class IgcChipComponent extends SizableMixin(
   }
 
   protected handleRemove(e: Event) {
+    if (
+      e.type === 'keyup' &&
+      ![' ', 'enter'].includes((e as KeyboardEvent).key.toLowerCase())
+    ) {
+      return;
+    }
     this.emitEvent('igcRemove');
     e.stopPropagation();
   }
```

Comment:
```suggestion
  protected handleRemove(e: Event) {
    this.emitEvent('igcRemove');
    e.stopPropagation();
  }
  
  protected handleKeyup(e: KeyboardEvent) {
    if (/\s|enter/i.test(e.key)) {
      this.handleRemove(e);
    }
  }
```
A bit cleaner IMO

Reply:
Thought regex is evil ^^

Reply:
It is a tool. It can be used for both good and evil 😄